src/Shell.hs

```diff
@@ -40,6 +59,18 @@ setPath p = do
   stRef <- ask
   modifyIORef stRef $ \st -> st {shellStPath = p}
 
+setVar :: String -> Val -> Shell ()
+setVar name val = do
+  stRef <- ask
+  modifyIORef stRef $ \st -> st {shellStEnv = Map.insert name val $ shellStEnv st}
+
+getVar :: String -> Shell Val
+getVar name = do
+ stRef <- ask
+ st <- readIORef stRef
+ let mt = fromJust $ Map.lookup name $ shellStEnv st --TODO: handle this potential error. We should implement error handling...
```

Comment:
Undeclared variable should not crash the program - either show an error to the user or assume it's empty string - bash does this.